mirgecom/simutil.py

```diff
@@ -425,3 +432,391 @@ def create_parallel_grid(comm, generate_grid):
+
+    # compare Geometry:
+    geo1 = file_reader1.get_geometry()
+    geo2 = file_reader2.get_geometry()
+
+    # check GeometryType
+    if geo1.get("GeometryType") != geo2.get("GeometryType"):
+        print("File 1:", geo1.get("GeometryType"), "\n",
+              "File 2:", geo2.get("GeometryType"))
+        raise ValueError("Fidelity test failed: Mismatched geometry type")
+
+    # check number of node values
+    nodes1 = file_reader1.read_data_item(tuple(geo1)[0])
+    nodes2 = file_reader2.read_data_item(tuple(geo2)[0])
+
+    nodes1 = np.array(nodes1)
+    nodes2 = np.array(nodes2)
+
+    if nodes1.shape != nodes2.shape:
+        print("File 1:", nodes1.shape, "\n", "File 2:", nodes2.shape)
+        raise ValueError("Fidelity test failed: Mismatched nodes count")
+
+    if not np.allclose(nodes1, nodes2, atol=tolerance):
+        print("Tolerance:", tolerance)
+        raise ValueError("Fidelity test failed: Mismatched node values with "
+                         "given tolerance")
+
+    # compare other Attributes:
+    for i in range(len(file_reader1.uniform_grid)):
+        curr_cell1 = file_reader1.uniform_grid[i]
+        curr_cell2 = file_reader2.uniform_grid[i]
+
+        # skip already checked cells
+        if curr_cell1.tag == "Geometry" or curr_cell1.tag == "Topology":
+            continue
+
+        # check AttributeType
+        if curr_cell1.get("AttributeType") != curr_cell2.get("AttributeType"):
+            print("File 1:", curr_cell1.get("AttributeType"), "\n",
+                  "File 2:", curr_cell2.get("AttributeType"))
+            raise ValueError("Fidelity test failed: Mismatched cell type")
+
+        # check Attribtue name
+        if curr_cell1.get("Name") != curr_cell2.get("Name"):
+            print("File 1:", curr_cell1.get("Name"), "\n",
+                  "File 2:", curr_cell2.get("Name"))
+            raise ValueError("Fidelity test failed: Mismatched cell name")
+
+        # check number of Attribute values
+        values1 = file_reader1.read_data_item(tuple(curr_cell1)[0])
+        values2 = file_reader2.read_data_item(tuple(curr_cell2)[0])
+
+        if len(values1) != len(values2):
+            print("File 1,", curr_cell1.get("Name"), ":", len(values1), "\n",
+                  "File 2,", curr_cell2.get("Name"), ":", len(values2))
+            raise ValueError("Fidelity test failed: Mismatched data values count")
+
+        # check values w/in tolerance
+        for i in range(len(values1)):
+            if abs(values1[i] - values2[i]) > tolerance:
+                print("Tolerance:", tolerance, "\n", "Cell:", curr_cell1.get("Name"))
+                raise ValueError("Fidelity test failed: Mismatched data values "
+                                 "with given tolerance")
+
+    print("XDMF Fidelity test completed successfully with tolerance", tolerance)
+
+
+def compare_files_hdf5(first_file: str, second_file: str, tolerance: float = 1e-12):
+    """Compare files of hdf5 type.
+
+    .. note::
```

Comment:
I think this line might not be necessary (same for the other functions)

```suggestion
    .. note::
```